.github/workflows/build-and-publish.yaml

```diff
@@ -0,0 +1,22 @@
+name: Build and Publish
+
+on:
+  pull_request:
```

Comment:
I don't see the point of running this workflow when a PR is opened. It is already triggered by the corresponding push event.